### PR TITLE
Use other prompt (prompt_dsn) when connecting using --dsn parameter (attempt 2)

### PR DESCRIPTION
**Pull request by msztolcman**
## Description
It's related to other PR: https://github.com/dbcli/pgcli/pull/815

And it's a result of discussion with @j-bennet (https://github.com/dbcli/pgcli/pull/815#issuecomment-349180420)

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by j-bennet** [comment]
@msztolcman Thank you for the PR! I want to try it out before merging, but looks good so far. I should be able to review tonight.

File: pgcli/main.py [hunk 1]
```diff
@@ -819,6 +830,7 @@ def get_last_query(self):
         default=False,
         help='Skip intro on startup and goodbye on exit.')
 @click.option('--prompt', help='Prompt format (Default: "\\u@\\h:\\d> ").')
+@click.option('--prompt-dsn', help='Prompt format when DSN alias is used (Default: "\\dsn> ").')
```
Comment (j-bennet) [inline pgcli/main.py]: You named it `'\\dsn_alias'` elsewhere, but `'\\dsn'` here.

Also, I don't see you actually setting the default.

File: pgcli/pgclirc [hunk 2]
```diff
@@ -111,10 +111,15 @@ row_limit = 1000
 less_chatty = False
 
 # Postgres prompt
+# \t - Current date and time
 # \u - Username
 # \h - Hostname of the server
 # \d - Database name
+# \p - Database port
```
Comment (j-bennet) [inline pgcli/pgclirc]: Thanks for adding these!

File: pgcli/main.py (hunk 1 above)
Comment (msztolcman) [reply pgcli/main.py]: Yep, it's artifacts from previous version, sorry. About default: yep, it's just using default from "normal" prompt.
Fixed now :)

File: pgcli/main.py [hunk 3]
```diff
@@ -819,6 +830,7 @@ def get_last_query(self):
         default=False,
         help='Skip intro on startup and goodbye on exit.')
 @click.option('--prompt', help='Prompt format (Default: "\\u@\\h:\\d> ").')
+@click.option('--prompt-dsn', help='Prompt format for connections using DSN aliases (default to same as for --prompt option).')
```
Comment (j-bennet) [inline pgcli/main.py]: Perhaps just say `Default: "\\u@\\h:\\d>`?

Comment (msztolcman) [reply pgcli/main.py]: If you are connecting using -D option, and specify --prompt, but not --prompt-dsn, then your prompt is not equal to "\\u@\\h:\\d>".
If you prefer version with explicit format I will change this, just explaining my point of view :)

Comment (j-bennet) [reply pgcli/main.py]: Well, in this docstring, you're only explaining the default, which is `\\u@\\h:\\d`. No need to make it more complicated by explaining all the use cases.

Comment (j-bennet) [reply pgcli/main.py]: @msztolcman So yes or no? :)

File: changelog.rst [hunk 4]
```diff
@@ -1,5 +1,6 @@
 Upcoming
 ========
+* Use other prompt (prompt_dsn) when connecting using --dsn parameter
```
Comment (j-bennet) [inline changelog.rst]: Oh, please add the `(Thanks: `Marcin Sztolcman`_)` part!

File: pgcli/main.py (hunk 3 above)
Comment (msztolcman) [reply pgcli/main.py]: @j-bennet Yes and done :)

**Comment by j-bennet** [comment]
@msztolcman You're missing the reference link in changelog.rst (that's why the build breaks in travis). Please add that:
```
.. _`Marcin Sztolcman`: https://github.com/msztolcman
```

**Comment by j-bennet** [comment]
All good now. Merging!

@msztolcman Thank you for sticking with it, despite all the  changes. 🍫

**Pull request merged**